corehq/util/metrics/datadog.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import List
+
+from django.conf import settings
+
+from corehq.util.datadog.utils import bucket_value
+from corehq.util.metrics.metrics import HqMetrics
+from datadog.dogstatsd.base import DogStatsd
+
+datadog_logger = logging.getLogger('datadog')
+
+COMMON_TAGS = ['environment:{}'.format(settings.SERVER_ENVIRONMENT)]
+
+statsd = DogStatsd(constant_tags=COMMON_TAGS)
+
+
+class DatadogMetrics(HqMetrics):
+    def initialize(self):
+        if not settings.DATADOG_API_KEY or not settings.DATADOG_APP_KEY:
+            raise Exception("Datadog not configured. Set DATADOG_API_KEY and DATADOG_APP_KEY in settings.")
```

Comment:
Could add something about removing `DatadogMetrics` from `settings.METRICS_PROVIDERS` to make it clear there's another way forward if this is blocking environment setup.